src/exceedances/job/views/exceedance.py

```diff
@@ -0,0 +1,387 @@
+                records[i]['backgroundColor'] = 'purple'
+            elif records[i]['characteristics']['no2Conc']['value'] > 1249 and records[i]['characteristics']['no2Conc']['value'] <= 2049:
+                records[i]['backgroundColor'] = 'maroon'
+            else:
+                records[i]['backgroundColor'] = 'gray'
+        return records
+    else:
+        projection = {'_id': 0, 'time': 1,
+                      'characteristics.pm2_5ConcMass.value': 1}
+        if frequency == 'hourly':
+            records = list(
+                db.device_hourly_measurements.find(query, projection))
+        elif frequency == 'daily':
+            records = list(
+                db.device_daily_measurements.find(query, projection))
+        else:
+            records = list(db.device_raw_measurements.find(query, projection))
+        for i in range(len(records)):
+            if records[i]['characteristics']['pm2_5ConcMass']['value'] > 0.0 and records[i]['characteristics']['pm2_5ConcMass']['value'] <= 12.0:
+                records[i]['backgroundColor'] = 'green'
+            elif records[i]['characteristics']['pm2_5ConcMass']['value'] > 12.0 and records[i]['characteristics']['pm2_5ConcMass']['value'] <= 35.4:
+                records[i]['backgroundColor'] = 'yellow'
+            elif records[i]['characteristics']['pm2_5ConcMass']['value'] > 35.4 and records[i]['characteristics']['pm2_5ConcMass']['value'] <= 55.4:
+                records[i]['backgroundColor'] = 'orange'
+            elif records[i]['characteristics']['pm2_5ConcMass']['value'] > 55.4 and records[i]['characteristics']['pm2_5ConcMass']['value'] <= 150.4:
+                records[i]['backgroundColor'] = 'red'
+            elif records[i]['characteristics']['pm2_5ConcMass']['value'] > 150.4 and records[i]['characteristics']['pm2_5ConcMass']['value'] <= 250.4:
+                records[i]['backgroundColor'] = 'purple'
+            elif records[i]['characteristics']['pm2_5ConcMass']['value'] > 250.4 and records[i]['characteristics']['pm2_5ConcMass']['value'] <= 500.4:
+                records[i]['backgroundColor'] = 'maroon'
+            else:
+                records[i]['backgroundColor'] = 'gray'
+        return records
+
+
+def calculate_exceedances_for_last_28_days():
+    pollutants = ['PM 2.5', 'PM 10', 'NO2']
+    standards = ['WHO', 'AQI']
+    exceedence_results = []
+    created_at = convert_date.str_to_date(
+        convert_date.date_to_str(datetime.now()))
+    for pollutant in pollutants:
+        for standard in standards:
+            exceedences_data = get_exceedances(pollutant, standard)
+            exceedence_record = {'pollutant': pollutant, 'standard': standard, 'exceedences': exceedences_data,
+                                 'created_at': created_at}
+            exceedence_results.append(exceedence_record)
+
+    print(exceedence_results)
+    save_device_daily_exceedences(exceedence_results)
```

Comment:
So afterwards, these exceedances are stored for future use. @ja-odur

Reply:
The purpose of this storage is mainly for optimisation purposes. So the question is: are these values needed for future analysis by the data scientists or we can just cache them and expire them at a predetermined time. @ja-odur

Reply:
Yes @Baalmart This has to be considered